packages/common/src/ui/components/inputs/CurrencyInput/CurrencyInput.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FC, useEffect } from 'react';
+import { styled } from '../../../../styles';
+import RCInput, {
+  CurrencyInputProps as RCInputProps,
+} from 'react-currency-input-field';
+
+interface CurrencyInputProps extends RCInputProps {
+  onChangeNumber: (value: number) => void;
+  maxWidth?: number;
+}
+
+// TODO: It would be nice if we were to just use the BasicTextInput or
+// another MUI text input rather than trying to recreate the style so that
+// it could stay in sync with style updated.
+const StyledCurrencyInput = styled(RCInput)(({ theme }) => ({
+  fontFamily: theme.typography.fontFamily,
+  height: 34.125,
+  borderRadius: '8px',
+  padding: '4px 8px',
+  backgroundColor: '#f2f2f5',
+  textAlign: 'right',
+  border: 'none',
+  '&:focus': {
+    outline: 'none',
+  },
+}));
+
+// TODO: I think the implementation of this could be swapped out by making a
+// headless component that uses something like currency.js in a custom hook
+// to manage the currency value. Then you could use whatever input component
+// you liked. That would just take a little more time than I have right now!
+export const CurrencyInput: FC<CurrencyInputProps> = ({
+  allowNegativeValue = false,
+  prefix = '$',
```

Comment:
I was wondering about the `$` sign somewhere else and was wondering how to get the correct locale for `intl` to format the currency. We are only using `en` which isn't enough. Perhaps though, we could just pull it from browser settings - the `i18n` returns full browser locale (such as `en-NZ`) and that would work ( although IIRC `en-NZ` wasn't supported.. there were a bunch of locales that intl falls down on. that aside, the approach might work 🤔  )

Reply:
I think I was thinking similarly. Use something like `useMoney()` and also the `useFormatNumber()` (I should have been using it for all decimals actually, oops) - currency could use the `useI18N` hook to get the current language and use that..

Reply:
Ah. I just clicked that language and currency aren't a one to one relationship so maybe that doesn't work 🤔